src/util/permissionIdentities.tsx

```diff
@@ -0,0 +1,183 @@
+import { LxdGroup, LxdIdentity } from "types/permissions";
+
+export type IdentitiesOrGroupsChangeSummary = Record<
+  string,
+  { added: Set<string>; removed: Set<string>; name: string }
+>;
+
+export const getIdentityIdsForGroup = (group: LxdGroup) => {
+  const oidcIdentityIds = group?.identities?.oidc || [];
+  const tlsIdentityIds = group?.identities?.tls || [];
+  const allIdentityIds = oidcIdentityIds.concat(tlsIdentityIds);
+  return allIdentityIds;
+};
+
+export const encodeIdentityNameForUrl = (name: string) => {
+  // having "." in the url results in 404 errors when refreshing the page
+  return encodeURIComponent(name.replaceAll(".", "#"));
+};
+
+export const decodeIdentityNameFromUrl = (name: string) => {
+  return decodeURIComponent(name).replaceAll("#", ".");
+};
+
+// Given a set of lxd groups and some identities
+// Generate a subset of those groups that's allocated to all identities
+// Generate a subset of those groups that's allocated to some identities
+export const getCurrentGroupsForIdentities = (
```

Comment:
```suggestion
export const getGroupsForIdentities = (
```
Easier to read and the "current" doesn't give much extra info?